### PR TITLE
Updated per library style guides.

### PR DESCRIPTION
**Pull request by dcbenito**
Signed-off-by: Benito, Dylan <dylan.benito@intel.com>

### CODE REVIEW

File: documentation/library_guide/index.rst [hunk 1]
```diff
@@ -1,10 +1,13 @@
-|onedpl_long|
-=================================
+|onedpl_long| Developer Guide and Reference
```
[the remark this answers is not in the document]
Comment (dcbenito) [reply documentation/library_guide/index.rst:1]: The recommendation was to update the name to Developer Guide and Reference so it aligns with the other library guides. I will continue to work on updating all instances, but for now I just fixed the top page.

File: documentation/library_guide/onedpl_gsg.rst [hunk 2]
```diff
@@ -17,8 +17,16 @@ For general information about |onedpl_short|, visit the `oneDPL GitHub* reposito
 or visit the `Intel® oneAPI DPC++ Library Guide <https://software.intel.com/content/www/us/en/develop/documentation/oneapi-dpcpp-library-guide/top.html>`_
 and the `Intel® oneAPI DPC++ Library main page <https://software.intel.com/content/www/us/en/develop/tools/oneapi/components/dpc-library.html>`_.
 
+Installation
+============
+
+Install link needs to go here.
```
[the remark this answers is not in the document]
Comment (dcbenito) [reply documentation/library_guide/onedpl_gsg.rst:23]: Perfect. I will update Before You Begin to Installation.

**Comment by dcbenito** [comment]
I have updated what I can take care of today. I will continue to update the doc name next week. Please review then merge.

**Review by dcbenito — COMMENTED** [review]
LGTM

File: documentation/library_guide/index.rst [hunk 3]
```diff
@@ -16,7 +19,7 @@ efforts across devices for high performance parallel applications.
    :maxdepth: 2
    :caption: Library Guide
 
-   overview
+   introduction
```
Comment (akukanov) [inline documentation/library_guide/index.rst:22]: Is the file renamed, or is a new file introduced? GitHub shows the diff as if a new file was added, yet the old file also remained intact. Let's make sure that there is only `introduction.rst` left when the patch is applied.

Comment (dcbenito) [reply documentation/library_guide/index.rst:22]: Overview should have been renamed to Introduction. Good idea to check so only intro is left.

**Comment by dcbenito** [comment]
Can @akukanov or @ValentinaKats merge this branch? I need to upload for IDZ and would like the most up to date version. I have to upload by EOD Friday this week.

**Review by akukanov — APPROVED** [review]

**Pull request merged**